src/renderer/lib/hooks/useReplays.ts

```diff
@@ -26,6 +31,11 @@ type StoreState = {
     total: number | null;
     fileResult: FileResult | null;
   };
+  // Can be incremented to force ReplayBrowser to re-render. This is useful
+  // because we can significantly improve performance by updating files in place
+  // (without copying the map), but this will not normally trigger a render
+  // because React does not recognize it as a state change.
+  forceRender: number;
```

Comment:
this feels like an anti-pattern in react, avoiding this kind of thing would be ideal.

Reply:
The problem is that doing a normal state update causes the entire files map to be copied, which becomes incredibly slow when this is being done for every update. I tried to implement this normally again tonight, trying two different approaches that I hoped would be a more clever, but it was still far too slow. First I tried using Immer, this was the slowest by far, I believe because it makes a deep copy for the draft state. Then I tried using Zustand alone, writing the code to intentionally avoid the deep copy, but I still had to make a shallow copy of the map and it was still slow.

The root of the issue as far as I know is React's insistence on immutable data. If you have an immutable map of 12k files (which is the number of replay files in my folder right now) and update one of those entries, best case scenario you have to copy 12k references in addition to the actual data you are updating. Do that 12k times, once for each entry as the files finish loading, and you're copying 144,000,000 references unnecessarily. This is not trivial.

The code above is the only solution that I have found that maintains good performance: Trick React by update the map in place, avoiding any unnecessary copies, and then updating the signal variable to tell React to start a new render.

If you have other ideas I'm willing to try them out.